### DISCUSSION TITLE
Writing JS tests

### DISCUSSION THREAD

**Discussion by tpluscode** [restcli/restcli#98] [Q&A]
Hello. Is there a more comprehensive documentation or detailed examples what is possible with the JS test feature? More specifically, I was thinking at least of 

- API doc for the `client` object used in tests
- how does one run tests in external JS files
   - I saw the `test_script.js` file in sample repo but not how it's used   
- is it possible to import external libraries in tests?
   - related, is ESM supported

**Comment by quangson91** [comment]
Hi @tpluscode 
Currently, the restcli supports JS test feature.
There are two ways to use JS test script.
1. Embed

https://github.dev/restcli/restcli/blob/8dcdc30d5fdade75714173c7f4350514d53b1d58/src/test/resources/requests/requests-share-var-between-files2.http#L1-L9

```
### Share variable file2
GET https://httpbin.org/get?id={{my_variable}}
Content-Type: application/json

> {%
client.test("Request executed successfully", function() {
  client.log("my_variable: " + client.global.get("my_variable"));
});
 %}
```

2. Use in the external file.

```
### Share variable file2
GET https://httpbin.org/get?id={{my_variable}}
Content-Type: application/json

> /path/to/test_script.js

```

> is it possible to import external libraries in tests?

This is not supported yet. But if you have any ideas, please share them here.
Since I am not a Javascript expert, I need more research.

Thank you for the feedback!

**Reply by tpluscode** [reply]
I opened https://github.com/restcli/restcli/issues/101
